src/TextInput/TextInputValidityState.js

```diff
@@ -82,6 +83,24 @@ export default class TextInputValidityState {
 		return {rangeUnderflow: (min !== "" && +value < +min), rangeOverflow: (max !== "" && +value > +max)};
 	}
 
+	/**
+	 * This method serves to supplement internal validation to determine whether a float qualifies as an integer.
+	 * The internal validation for an input type="number" step="1" doesn't consistently cover all cases.
+	 * For example, the number 2.00000001 might be considered an integer by the internal validator.
+	 * The aim is to widen this validation range. Now, the comparison for an integer works correctly when a number
+	 * has no more than 16 digits. However, if a number has 17 digits or more, this function returns incorrect result.
+	 * @return {{stepMismatch: boolean}}
+	 */
+	#isStepConstraintViolated() {
```

Comment:
Step can exist even if we expect numbers with decimal point. I don't think we can add this validation "by default" if the step is specified.

Instead I'd create an implementation of the validator (what you call an "external" validator at the moment) and place it inside ui-essentials. And then the client code may decide to add it as a validator. It's just in this case the client code won't have to create its own implementation - it already exists in the ui-essential.